lld/ELF/Driver.cpp

```diff
@@ -1629,7 +1667,6 @@ static void readConfigs(Ctx &ctx, opt::InputArgList &args) {
       std::make_pair("bti-report", &ctx.arg.zBtiReport),
       std::make_pair("cet-report", &ctx.arg.zCetReport),
       std::make_pair("execute-only-report", &ctx.arg.zExecuteOnlyReport),
-      std::make_pair("gcs-report", &ctx.arg.zGcsReport),
       std::make_pair("pauth-report", &ctx.arg.zPauthReport)};
```

Comment:
I think it's better to handle gcs-report and gcs-report-dynamic here. We can add a special case to track whether gcs-report-dynamic has been specified

Reply:
I disagree. The is significant duplication and code bloat due to none/warning/error parsing.
You could add
```
if (option.first == "gcs-report-dynamic")
  reportDynamicSpecified = true
```

then after the for loop,
```
  if (!reportDynamicDefined && ctx.arg.zGcsReport != ReportPolicy::None &&
      ctx.arg.zGcsReportDynamic == ReportPolicy::None)
    ctx.arg.zGcsReportDynamic = ReportPolicy::Warning;
```